recipes/dcmtk/all/conanfile.py

```diff
@@ -118,12 +118,17 @@ def validate(self):
            self.settings.os == "Macos" and self.settings.arch == "armv8":
             # FIXME: Probable issue with flags, build includes header 'mmintrin.h'
             raise ConanInvalidConfiguration("Cross building to Macos M1 is not supported (yet)")
+        if hasattr(self, "settings_build") and cross_building(self) and \
+           self.settings.os == "Macos" and self.settings.arch == "x86_64" and not can_run(self):
+            raise ConanInvalidConfiguration("Cross building to macOS x86_64 is only supported from macOS arm64 using rosetta with 'tools.build.cross_building:can_run=True'")
```

Comment:
With the new macOS builders in CI using an arm64 build profile to build for x86_64 host, this recipe will now raise a ConanInvalidConfiguration exception.

Will the config be updated with `tools.build.cross_building:can_run=True` ?
If not, is there a simple workaround recommended ?

Reply:
@uilianries @danimtb Tagging you as you were the most visibly involved with the M2 upgrade.

Can the M2 machines run both `armv8` and `x86_64` executables? And even if they can run both, should we still probably try to use the `armv8` versions of the executables?

Reply:
> And even if they can run both, should we still probably try to use the armv8 versions of the executables?

In this specific instance, there's no `armv8` version of the executable as it only there to allow some `try_run` calls in the CMake file.
Having `tools.build.cross_building:can_run=True` would also mean that tests could run entirely which is not the case anymore for macOS x86_64 builds.

I could add a specific `_can_run` property to the recipe in order not to require  `tools.build.cross_building:can_run=True` if rosetta is available.

Reply:
> Can the M2 machines run both armv8 and x86_64 executables? And even if they can run both, should we still probably try to use the armv8 versions of the executables?

Yes, M2 can build and run both amrv8 and x86_64. As M2 can run anything, there is no favorite target when running, when a Mac machine is available, it will run tests or build binaries as needed.

Reply:
> it will run tests

it won't run any executable built by the tests and protected with `if can_run(self):` while it could. That's what I meant by "tests could run entirely which is not the case anymore for macOS x86_64 builds".